lib/checkleakautovar.cpp

```diff
@@ -740,6 +763,9 @@ void CheckLeakAutoVar::functionCall(const Token *tokName, const Token *tokOpenin
     // Ignore function call?
     if (mSettings->library.isLeakIgnore(tokName->str()))
         return;
+    const Library::AllocFunc* g = mSettings->library.realloc(tokName);
+    if (g)
```

Comment:
I suggest that you put the function call in the condition. No variable is needed here.